src/main/java/qna/domain/answer/Answer.java

```diff
@@ -79,8 +81,16 @@ public boolean isDeleted() {
         return deleted;
     }
 
-    public void setDeleted(boolean deleted) {
-        this.deleted = deleted;
+    public DeleteHistory delete() {
+        if (deleted) {
+            throw new AlreadyDeletedException("이미 삭제된 답입니다.");
+        }
+        this.deleted = true;
+        return toDeleteHistory();
```

Comment:
```suggestion
        return DeleteHistory.answer(id, writer);
```

와 같이 정적 팩토리 메서드를 활용하는 방법도 있습니다 ㅎㅎ